org.eclipse.jdt.debug/model/org/eclipse/jdt/debug/core/IJavaStackFrame.java

```diff
@@ -533,4 +570,18 @@ public IJavaVariable findVariable(String variableName)
 	 * @since 3.3
 	 */
 	public void forceReturn(IJavaValue value) throws DebugException;
+
+	/**
+	 * @since 3.22
+	 * @param category
+	 *            the new category of the stack frame.
+	 */
+	public void setCategory(Category category);
```

Comment:
can it somehow avoided to have such setter? it is not obvious to me if such setter should `fireChangeEvent(DebugEvent.STATE);`

Reply:
Somehow we need to store the information how that frame was categorized - even if the process is not slow, it needs to lookup the source, and check which classpath entry contains that, etc.
In theory, we could have a map from IJavaStackFrame -> Category in some singleton, but it's unclear how we can detect, if that stack frame was not accessible anymore, so maybe a WeakHashMap could work, but it felt too complicated and error prone.
 This setter is only called once - when we first want to display the frame, so firing a change event should not needed - at least it seems to work fine without it.